src/logic/calculate.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-unused-vars */
+import operate from './operate';
+/* eslint-enable no-unused-vars */
+
+const calculate = (calcData, buttonName) => {
+  let { next, total, operation } = calcData;
+
+  switch (buttonName) {
+    case '+/-':
+      total *= -1;
+      next *= -1;
+      break;
+    case 'AC':
+      total = null;
+      next = null;
+      break;
+    case '%':
+      operation = '%';
+      break;
+    case '.':
```

Comment:
if the `buttonName`  = `.` we should return `0.` just like a normal calculator on our devices.